stl/inc/future

```diff
@@ -1043,27 +1043,21 @@ _NODISCARD inline shared_future<void> future<void>::share() noexcept {
 template <class _Ty>
 class _Promise {
 public:
-    _Promise(_Associated_state<_Ty>* _State_ptr) : _State(_State_ptr, false), _Future_retrieved(false) {}
+    _Promise(_Associated_state<_Ty>* _State_ptr) noexcept : _State(_State_ptr, false), _Future_retrieved(false) {}
 
-    _Promise(_Promise&& _Other) : _State(_STD move(_Other._State)), _Future_retrieved(_Other._Future_retrieved) {}
+    _Promise(_Promise&&) = default;

```

Comment:
Note that this does not make `_Promise<T>` trivially moveable since `_State_manager<T>` is never trivially moveable (and therefore same for `promise<T>` and `packaged_task<T>`)

Reply:
Thanks, I think that makes this ABI-safe. (This is very close to the dangerous condition that I encountered in `tuple<>`'s copy ctor, though.)